api/src/main/java/me/lokka30/treasury/api/common/misc/FutureHelper.java

```diff
@@ -0,0 +1,54 @@
+package me.lokka30.treasury.api.common.misc;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Function;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * A utility class to help with {@link CompletableFuture} stuff
+ *
+ * @author MrIvanPlays
+ * @since 1.2.1
+ */
+public class FutureHelper {
+
+    /**
+     * Map, join, filter - this is what the name of the method is.
+     * Maps all the futures holding data type A to return a collection of data type B
+     * via the specified {@code mapper} {@link Function} and filters them via the specified
+     * filter function.
+     * <br><b>WARNING: This is a thread unsafe method. ASYNCHRONOUS CALL MANDATORY!!!!</b>
```

Comment:
Would you like me to fix this? It is entirely possible to turn a collection of futures into a future of a collection, without using `CompletableFuture#join`.

As the method currently stands, it seems non-sensical to return a `CompletableFuture<Collection<B>>` considering this future will always be complete by the time the method returns. It might as well return `Collection<B>`.

Reply:
That's actually a good point

Reply:
Joining misses the purpose of a completable future.

Reply:
See #196